coral/src/app/features/approvals/components/RequestDetailsModal.test.tsx

```diff
@@ -0,0 +1,74 @@
+import { cleanup, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import RequestDetailsModal from "src/app/features/approvals/components/RequestDetailsModal";
+
+const baseProps = {
+  onClose: jest.fn(),
+  onApprove: jest.fn(),
+  onReject: jest.fn(),
+};
+
+describe.only("RequestDetailsModal.test", () => {
+  afterEach(cleanup);
+
+  it("renders a Modal with correct elements (isLoading false)", () => {
+    render(
+      <RequestDetailsModal {...baseProps} isLoading={false}>
+        <div>content</div>
+      </RequestDetailsModal>
+    );
+
+    expect(
+      screen.getByRole("heading", { name: "Request details" })
+    ).toBeVisible();
+    expect(screen.getByText("content")).toBeVisible();
+    expect(screen.getByRole("button", { name: "Close modal" })).toBeVisible();
+    expect(screen.getByRole("button", { name: "Close modal" })).toBeVisible();
+    expect(screen.getByRole("button", { name: "Approve" })).toBeEnabled();
+    expect(screen.getByRole("button", { name: "Approve" })).toBeEnabled();
+    expect(screen.getByRole("button", { name: "Reject" })).toBeVisible();
+    expect(screen.getByRole("button", { name: "Reject" })).toBeEnabled();
+  });
+
+  it("renders a Modal with correct elements (isLoading true)", () => {
+    render(
+      <RequestDetailsModal {...baseProps} isLoading={true}>
+        <div>content</div>
+      </RequestDetailsModal>
+    );
+
+    expect(
+      screen.getByRole("heading", { name: "Request details" })
+    ).toBeVisible();
+    expect(screen.getByText("content")).toBeVisible();
+    expect(screen.getByRole("button", { name: "Close modal" })).toBeVisible();
+    expect(screen.getByRole("button", { name: "Close modal" })).toBeVisible();
+    expect(screen.getByRole("button", { name: "Approve" })).toBeVisible();
+    expect(screen.getByRole("button", { name: "Approve" })).toBeDisabled();
+    expect(screen.getByRole("button", { name: "Reject" })).toBeVisible();
+    expect(screen.getByRole("button", { name: "Reject" })).toBeDisabled();
+  });
+
+  it("calls correct functions when clicking buttons", async () => {
```

Comment:
Same as above -> separate blocks with one test case. It's more readable, faster to debug.

Reply:
[ce76e91](https://github.com/aiven/klaw/pull/611/commits/ce76e91b9d29800e09573ccd2a40936cb6743a68)